uo_sys_framework/ACREModule/cfgFunctions/common/fn_addRack.sqf

```diff
@@ -0,0 +1,148 @@
+	_radioRemovable = false;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING TO: %1", _radioRemovable]) call UO_FW_FNC_DebugMessage;
+};
+
+_args pushBackUnique _radioRemovable;
+
+if(_mountRadio) then {
+	switch (_rackType) do {
+		case "ACRE_SEM90": {_mountedRadio = "ACRE_SEM70";};
+		case "ACRE_VRC103": {_mountedRadio = "ACRE_PRC117F";};
+		case "ACRE_VRC110": {_mountedRadio = "ACRE_PRC152";};
+		case "ACRE_VRC111": {_mountedRadio = "ACRE_PRC148";};
+		case "ACRE_VRC64": {_mountedRadio = "ACRE_PRC77";};
+		default {
+			(format ["(ACRE_SETUP_MODULE) [ERR]: NFB ERROR ENCOUNTERED"]) call UO_FW_FNC_DebugMessage;
+			(format ["(ACRE_SETUP_MODULE) [ERR]: VERIFY YOUR RACK CLASSNAME: %1", _rackType]) call UO_FW_FNC_DebugMessage;
+		};
+	};
+} else {_mountedRadio = "";};
+
+_args pushBackUnique _mountedRadio;
+
+if(isNil "_rackNameLong" || isNil "_rackNameShort") then {
+	(format ["(ACRE_SETUP_MODULE) [ERR]: A RACKNAME IS NIL, LONG: %1 SHORT: %2", _rackNameLong, _rackNameShort]) call UO_FW_FNC_DebugMessage;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING TO LONGNAME: ""DEFAULT RACK NAME: FIX ME"" AND SHORTNAME: ""FIX"""]) call UO_FW_FNC_DebugMessage;
+	_rackNameLong = "DEFAULT RACK NAME: FIX ME";
+	_rackNameShort = "FIX";
+};
+
+_args pushBackUnique _rackNameLong;
+_args pushBackUnique _rackNameShort;
+
+if(_whitelistedPositions isEqualTo [] && _blacklistedPositions isEqualTo []) then {
+	(format ["(ACRE_SETUP_MODULE) [ERR]: NO POSITIONS HAVE BEEN WHITELISTED OR BLACKLISTED"]) call UO_FW_FNC_DebugMessage;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING TO 'inside' and 'all'"]) call UO_FW_FNC_DebugMessage;
+	_whitelistedPositions = ["inside", "all"];
+} else {
+	if(_whitelistedPositions isEqualTo _blacklistedPositions) then {
+		(format ["(ACRE_SETUP_MODULE) [ERR]: ALL POSITIONS ARE WHITELISTED AND BLACKLISTED"]) call UO_FW_FNC_DebugMessage;
+		(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING WHITELISTED POSITIONS TO: 'inside'"]) call UO_FW_FNC_DebugMessage;
+		_whitelistedPositions = ["inside"];
+		_blacklistedPositions = [];
+	};
+};
+
+_args pushBackUnique _whitelistedPositions;
+_args pushBackUnique _blacklistedPositions;
+
+_args pushBackUnique _faction;
+
+_statement = {
+	if(!(isNil "_faction") && {UO_FW_ACRE_SCRAMBLE_ENABLED}) then {
+		private _faction_i = 3;
+		switch (_faction) do {
+			case west: {_faction_i = 0};
+			case east: {_faction_i = 1};
+			case independent: {_faction_i = 2};
+			case civilian: {_faction_i = 3};
+			default {_faction_i = 3;};
+		};
+		private _preset = _radioScramblePresets select _faction_i;
+		[_object, _preset] call acre_api_fnc_setVehicleRacksPreset;
+	};
+
+	[_object, {}] call acre_api_fnc_initVehicleRacks;
+	[_object, [_rackType, _rackNameLong, _rackNameShort, _radioRemovable, _whitelistedPositions, _blacklistedPositions, _mountedRadio, [], []], false, {}] call acre_api_fnc_addRackToVehicle;
```

Comment:
Are these vars defined or do you need to `params` them within the `CBA_fnc_waitUntilAndExecute` since it creates a new scope?

Reply:
Will need to params. Didn't realize I hadn't done so already.

Reply:
You might not need to create the `_args` array using the `pushback` commands, you could just pass it as a args array with the elements referenced in it. eg. `[_object, _rackType, _rackNameLong, _rackNameShort, _radioRemovable, _whitelistedPositions, _blacklistedPositions, _mountedRadio]` as your `CBA_fnc_waitUntilAndExecute` arg param. Then just `params` the same array by copying it into the statement block and wrapping each of the elements in `"`, that way you know exactly what is being passed.

Reply:
I'm sticking to using _args as this will be used alongside to some custom macros for the UOFW which are CBA compliant in order to give an example of a fully CBA compliant module.

Reply:
What? Why is that a CBA/ACE standard? Seems like they use defined var elements in their args params regularly. 

https://github.com/acemod/ACE3/blob/59aa38185a7fec3200028a73e75c4ff424ae3809/addons/interaction/functions/fnc_pullOutBody.sqf#L62

or 

https://github.com/acemod/ACE3/blob/59aa38185a7fec3200028a73e75c4ff424ae3809/addons/medical/functions/fnc_addDamageToUnit.sqf#L56

Reply:
Miscommunication. I intended it to be an example of the syntax used for calling the function rather than stating that what I was doing was 'standard' in any way. Apologies.